var/spack/repos/builtin/packages/curl/package.py

```diff
@@ -146,31 +146,53 @@ class Curl(NMakePackage, AutotoolsPackage):
 
     @classmethod
     def determine_version(cls, exe):
-        curl = Executable(exe)
-        output = curl("--version", output=str, error=str)
-        match = re.match(r"curl ([\d.]+)", output)
+        curl_config = Executable(exe)
+        output = curl_config("--version", output=str, error=str)
+        match = re.match(r"libcurl ([\d.]+)", output)
         return match.group(1) if match else None
 
     @classmethod
     def determine_variants(cls, exes, version):
         for exe in exes:
             variants = ""
             curl = Executable(exe)
-            output = curl("--version", output=str, error=str)
-            if "nghttp2" in output:
-                variants += "+nghttp2"
-            protocols_match = re.search(r"Protocols: (.*)\n", output)
-            if protocols_match:
-                protocols = protocols_match.group(1).strip().split(" ")
-                if "ldap" in protocols:
-                    variants += "+ldap"
-            features_match = re.search(r"Features: (.*)\n", output)
-            if features_match:
-                features = features_match.group(1).strip().split(" ")
-                if "GSS-API" in features:
-                    variants += "+gssapi"
-            # TODO: Determine TLS backend if needed.
-            # TODO: Determine more variants.
+            output = curl("--features", "--protocols", "--configure", "--ssl-backends", output=str)
```

Comment:
I think it would be safer to split this into output for the categories. If someone adds an explicit `--without-mbedtls`, then `"mbedtls" in output` will be True. The only thing saving us from that is capitalization since we check for `mbedTLS`. That seems quite fragile.

Reply:
My case for it is that it does work, and is more efficient than 4 individual fork+execs for each parameter.